File: Data Structures & Algorithms/buy-and-sell-crypto/submission-5.py
class Solution:
    def maxProfit(self, p: List[int]) -> int:
        res = 0
        l,r = 0,0
        while r < len(p):
            profit = p[r] - p[l]
            if profit > res:
                res = profit
            elif profit < 0:
                l = r
            r += 1
        return res
        

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.js
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    lengthOfLongestSubstring(s) {
        let res = 0;
        const sset = {}
        let l = 0;
        let r = 0;
        while(r < s.length){
            sset[s[r]] = (sset[s[r]]||0)+1
            while(sset[s[r]] > 1){
                sset[s[l]] -= 1
                l += 1
            }
            res = Math.max(res,r-l+1)
            r += 1
        }
        return res
    }
}
